operators/multiclusterobservability/api/v1beta2/multiclusterobservability_types.go

```diff
@@ -142,6 +148,24 @@ type PlatformIncidentDetectionSpec struct {
 	Enabled bool `json:"enabled,omitempty"`
 }
 
+// NamespaceRightSizingSpec defines the namespace right sizing capabilities.
+// +kubebuilder:object:generate=true
+// +kubebuilder:validation:Type=object
```

Comment:
I think you don't need the kubebuilder annotations here

Reply:
ACK